File: frontend/modules/selection-actions/android/src/main/java/expo/modules/selectionactions/SelectionActionsModule.kt
package expo.modules.selectionactions

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class SelectionActionsModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("SelectionActions")

    View(SelectableTranslateTextView::class) {
      Events("onTranslate")

      Prop("text") { view: SelectableTranslateTextView, value: String ->
        view.applyText(value)
      }

      Prop("fontSize") { view: SelectableTranslateTextView, value: Double? ->
        view.applyFontSize(value ?: 15.0)
      }

      Prop("color") { view: SelectableTranslateTextView, value: String? ->
        view.applyColorHex(value)
      }

      Prop("translateLabel") { view: SelectableTranslateTextView, value: String? ->
        view.translateLabel = value ?: "Translate"
      }
    }
  }
}

File: frontend/modules/selection-actions/ios/SelectionActionsModule.swift
import ExpoModulesCore

public class SelectionActionsModule: Module {
  public func definition() -> ModuleDefinition {
    Name("SelectionActions")

    View(SelectableTranslateTextView.self) {
      Events("onTranslate")

      Prop("text") { (view: SelectableTranslateTextView, value: String) in
        view.setText(value)
      }

      Prop("fontSize") { (view: SelectableTranslateTextView, value: Double?) in
        view.setFontSize(value ?? 15)
      }

      Prop("color") { (view: SelectableTranslateTextView, value: String?) in
        view.setTextColorHex(value)
      }

      Prop("translateLabel") { (view: SelectableTranslateTextView, value: String?) in
        view.setTranslateLabel(value ?? "Translate")
      }
    }
  }
}
